src/Input/LocalInputSource.php

```diff
@@ -93,42 +97,139 @@ public function isPDF(): bool
 
     /**
      * Counts the amount of pages in a PDF.
-     * To be implemented.
      *
-     * @return void
+     * @return integer
+     * @throws MindeePDFException Throws if the source pdf can't be properly processed.
+     * @throws MindeeSourceException Throws if the source isn't a pdf.
      */
-    public function countDocPages() // TODO: add PDF lib
+    public function countDocPages(): int
```

Comment:
It doesn't look like the page count is returned, unless I'm missing something.

Reply:
Yeah the syntax is [hella weird](https://manuals.setasign.com/fpdi-manual/v1/the-fpdi-class/#index-3-1), but it does return the page count.

Reply:
Such API, very clean. Much logic.